src/include/crowbar/complex.rb

```diff
@@ -469,20 +472,22 @@ def InitRepoURL(id)
     # initialize the value of repo table
     def InitReposTable(id)
       repo_items = []
-      @repos.each do |prod_name, platform|
-        platform.each do |repo_id, repo|
-          repo_items <<
-            Item(
-              Id("#{prod_name}|#{repo_id}"),
-              repo["name"] || repo_id,
-              repo["url"] || "",
-              (repo["ask_on_error"] || false) ?  UI.Glyph(:CheckMark) : " ",
-              prod_name
-            )
+      @repos.each do |platform, arches|
+        arches.each do |arch, repos|
+          repos.each do |repo_id, repo|
+            repo_items <<
+              Item(
+                Id("#{platform}|#{arch}|#{repo_id}"),
+                repo["name"] || repo_id,
+                repo["url"] || "",
+                (repo["ask_on_error"] || false) ?  UI.Glyph(:CheckMark) : " ",
+                "#{@platform2label[platform] || platform} (#{arch})"
+              )
+          end
         end
       end
       UI.ChangeWidget(Id(id), :Items, repo_items)
-      UI.ChangeWidget(Id(id), :CurrentItem, "#{@current_repo_platform}|#{@current_repo}") unless @current_repo.empty?
+      UI.ChangeWidget(Id(id), :CurrentItem, "#{@current_repo_platform}|#{@current_repo}|#{@current_repo_id}") unless @current_repo.empty?
```

Comment:
It works well, that's just minor UI glitch, probably only relevant in Qt anyway.